domain/infosync/tiflash_manager.go

```diff
@@ -386,21 +410,24 @@ func makeBaseRule() placement.TiFlashRule {
 }
 
 // MakeNewRule creates a pd rule for TiFlash.
-func MakeNewRule(ID int64, Count uint64, LocationLabels []string) *placement.TiFlashRule {
-	ruleID := fmt.Sprintf("table-%v-r", ID)
-	startKey := tablecodec.GenTableRecordPrefix(ID)
-	endKey := tablecodec.EncodeTablePrefix(ID + 1)
-	startKey = codec.EncodeBytes([]byte{}, startKey)
-	endKey = codec.EncodeBytes([]byte{}, endKey)
+func MakeNewRule(id int64, count uint64, locationLabels []string) placement.TiFlashRule {
+	ruleID := MakeRuleID(id)
+	startKey := tablecodec.GenTableRecordPrefix(id)
+	endKey := tablecodec.EncodeTablePrefix(id + 1)
 
 	ruleNew := makeBaseRule()
 	ruleNew.ID = ruleID
-	ruleNew.StartKeyHex = startKey.String()
-	ruleNew.EndKeyHex = endKey.String()
-	ruleNew.Count = int(Count)
-	ruleNew.LocationLabels = LocationLabels
+	ruleNew.StartKey = startKey
+	ruleNew.EndKey = endKey
+	ruleNew.Count = int(count)
+	ruleNew.LocationLabels = locationLabels
+
+	return ruleNew
+}
 
-	return &ruleNew
+// MakeRuleID creates a rule ID for TiFlash with given TableID.
```

Comment:
So this function is used for both API v1 and v2, and in v2 it will be decorated by `encodeRuleID`? If so, maybe we can elaborate comments here.

Reply:
OK